dns-metrics.schema.json

```diff
@@ -171,9 +171,160 @@
                             }
                         }
                     }
+                },
+                "conn_info": {
+                    "$ref": "#/$defs/conn_info"
                 }
             },
             "required": [ "since", "until" ]
+        },
+        "udp_conn": {
+            "title": "DNS Metric UDP Connection Information",
+            "description": "An object with the metrics around the UDP connections used to sent and received for the given stats object",
+            "type": "object",
+            "properties": {
+                "packets_sent": {
+                    "description": "The number of packets sent",
+                    "type": "integer"
+                },
+                "packets_received": {
+                    "description": "The number of packets received",
+                    "type": "integer"
+                },
+                "socket_errors": {
+                    "description": "The number of various socket errors that occurred",
+                    "type": "integer"
+                }
+            }
+        },
+        "tcp_conn": {
+            "$ref": "#/$defs/udp_conn",
+            "title": "DNS Metric TCP Connection Information",
+            "description": "An object with the metrics around the TCP connections used to sent and received for the given stats object",
+            "properties": {
+                "handshakes": {
+                    "description": "The number of handshakes performed",
+                    "type": "integer"
+                },
+                "handshakes_failed": {
+                    "description": "The number of failed handshakes",
+                    "type": "integer"
+                }
+            }
+        },
+        "conn_latency_stats": {
+            "type": "object",
+            "properties": {
+                "avg": {
+                    "description": "The average latency, in nanoseconds",
+                    "type": "integer"
+                },
+                "min": {
+                    "description": "The minimum latency, in nanoseconds",
+                    "type": "integer"
+                },
+                "max": {
+                    "description": "The maximum latency, in nanoseconds",
+                    "type": "integer"
+                },
+                "stddev": {
+                    "description": "The standard deviation for latency, in nanoseconds",
+                    "type": "integer"
+                },
+                "buckets": {
+                    "description": "??TODO??",
+                    "type": "array",
+                    "items": {
+                        "type": "array",
+                        "items": {
+                            "type": "integer"
+                        }
+                    }
+                }
+            }
+        },
```

Comment:
Maybe we can have a generic "latency" object in the schema and reference it from all other places? It should be the same for DNS queries as well, I think.

Reply:
Thread migrated to: https://github.com/DNS-OARC/dns-metrics/issues/12